src/Core/src/Hosting/Internal/MauiServiceProvider.cs

```diff
@@ -252,5 +258,9 @@ static List<Type> GetServiceBaseTypes(Type serviceType)
 
 			throw new InvalidOperationException($"Could not match any constructors for '{implementationType.Name}'.");
 		}
+
+		[UnconditionalSuppressMessage("Trimming", "IL2073", Justification = "We cannot specify DynamicallyAccessedMemberTypes.Interfaces on ParameterInfo.ParameterType.")]
```

Comment:
That's true - but what makes it OK to ignore the warning in this case?

Reply:
Yeah, similar question to: https://github.com/dotnet/maui/pull/3249#discussion_r755700416

If you have a method like `foo(ParameterInfo p)`, how do you specify `DynamicallyAccessedMemberTypes.Interfaces` on `p.ParameterType`?

Reply:
I don't think we have a way to fix this currently. Depends on the scenario. We might be able to add some new functionality but I would have to understand what this is for.

The general rule for these things in the linker is:
* It should not keep more things than necessary (or at least close to this)
* It must be reasonably efficient to implement (for example, requiring the linker to scan through all types in the app is not reasonable)
* The usage pattern must be statically analyzable - meaning it's not enough to come up with a way to tell linker what to keep, the linker should also be able to validate that the reflection calls used to access such things are correct.